test/peer-routing.spec.js

```diff
@@ -0,0 +1,63 @@
+/* eslint-env mocha */
+'use strict'
+
+const chai = require('chai')
+chai.use(require('dirty-chai'))
+const expect = chai.expect
+const parallel = require('async/parallel')
+const utils = require('./utils')
+const createNode = utils.createNode
+const _times = require('lodash.times')
+
+describe.only('.peerRouting', () => {
+  let nodeA
+  let nodeB
+  let nodeC
+  let nodeD
+  let nodeE
+
+  before((done) => {
+    const tasks = _times(5, () => (cb) => {
+      createNode('/ip4/0.0.0.0/tcp/0', { mdns: false }, (err, node) => {
+        expect(err).to.not.exist()
+        node.start((err) => cb(err, node))
+      })
+    })
+
+    parallel(tasks, (err, nodes) => {
+      expect(err).to.not.exist()
+      nodeA = nodes[0]
+      nodeB = nodes[1]
+      nodeC = nodes[2]
+      nodeD = nodes[3]
+      nodeE = nodes[4]
+
+      parallel([
+        (cb) => nodeA.dial(nodeB.peerInfo, cb),
+        (cb) => nodeB.dial(nodeC.peerInfo, cb),
+        (cb) => nodeC.dial(nodeD.peerInfo, cb),
+        (cb) => nodeD.dial(nodeE.peerInfo, cb),
+        (cb) => nodeE.dial(nodeA.peerInfo, cb)
+      ], done)
+    })
+  })
+
+  after((done) => {
+    parallel([
+      (cb) => nodeA.stop(cb),
+      (cb) => nodeB.stop(cb),
+      (cb) => nodeC.stop(cb),
+      (cb) => nodeD.stop(cb),
+      (cb) => nodeE.stop(cb)
+    ], done)
+  })
+
+  describe('ring', () => {
+    it('nodeA.dial by Id to node C', (done) => {
+      nodeA.dial(nodeC.peerInfo.id, done)
+    })
```

Comment:
I've added some delay in order to ensure that the network is stabilized (which should not be required) and now I'm getting a different error. The peerRouting.findPeer is returning undefined as the peer and not a error in case of peer not found, making swarm to fail.